src/main/scala/me/shadaj/scalapy/numpy/NDArray.scala

```diff
@@ -3,28 +3,30 @@ package me.shadaj.scalapy.numpy
 import me.shadaj.scalapy.py
 import me.shadaj.scalapy.py.{PyValue, Reader, Writer}
 
-class NDArray[T](val value: PyValue)(implicit reader: Reader[T]) extends py.Object with Seq[T] {
+class NDArray[+T](val value: PyValue)(implicit reader: Reader[T]) extends py.Object with Seq[T] {
```

Comment:
Hmmm but is `NDArray` actually covariant? AFAICS it gives access to the underlying data (https://numpy.org/doc/stable/reference/generated/numpy.ndarray.data.html#numpy.ndarray.data) which can be modified

Reply:
wait what? `data` gives a pointer to the beginning of the array, but I dont know how is it relevant. 

Aren't collections generally covariant? 
I mean NDArray[Int] <: NDArray[Any] should hold in my opinion. 

Maybe I should explain myself why I added this. 
I wanted to make `Model.fit(...)` arguments of type NDArray[Numeric] so you can train a network on Longs/Ints/Doubles and so on.  But I had hard time working with scala Numeric.

Reply:
> data gives a pointer to the beginning of the array, but I dont know how is it relevant.

The underlying data can be modified, rendering the entire collection **mutable** and hence not covariant (and not contravariant either). Also, even if `data` wasn't exposed in the way it is... aren't there a couple other ways to modify an existing `NDArray` anyway? `__setitem_`?

> Aren't collections generally covariant?
Whoops no... they're generally only covariant when immutable :/

Reply:
>Whoops no... they're generally only covariant when immutable :/
Ooooo ok.

ok agree NDArray is mutable. so it should stay invariant? 
But that way we would have to specify an exact type in python methods like `x: NDArray[Double] | NDArray[Int] | ...` ?

Reply:
I think the approach taken by Java for arrays might be useful for us

Reply:
They are covariant but check for unsound modifications and can throw ArrayStoreException

Reply:
ok thats good, but it throws this exception while setting value. NDArray has no way to set a value for now, so I will add a comment summarizing this discussion.  
How about that?

Reply:
Ok pls drop a comment referring to an issue in this forked repo maybe so that we don't lose track of the problem

Reply:
OK